nginx/setup.sh

```diff
@@ -1,82 +1,38 @@
 
-echo "Checking AWS credentials are valid"
-if [ -z "$1" ]; then
-	PROFILE=""
-else
-	PROFILE="--profile ${1}"
-fi
+# mkcert requires the JAVA_HOME envvar to be set to add the generated CA to Java's trust store
+# see https://github.com/FiloSottile/mkcert#supported-root-stores
+export JAVA_HOME=$(/usr/libexec/java_home)
 
-aws ${PROFILE} s3 ls ${S3_BUCKET} 1>/dev/null 2>&1
-if [[ $? -gt 0 ]]; then
-	echo "ERROR: You do not have access to the Identity AWS account. Re-run with ${0} <profile name> to use a different profile."
-	exit 2
-fi
+mkcert -install
 
-echo "Checking Nginx sites-enabled directory exists"
-ls "${nginxHome}/sites-enabled" 1>/dev/null 2>&1
-if [[ $? -gt 0 ]]; then
-	echo "ERROR: Missing ${nginxHome}/sites-enabled"
-    echo "Create sites-enabled directory and make sure it is included in your nginx.conf (usually in ${nginxHome}/nginx.conf):
-            http {
-                include       mime.types;
-                default_type  application/octet-stream;
-                # THIS IS WHAT YOU MUST ADD
-                include sites-enabled/*;
-            #..."
-	exit 1
-fi
+echo -e "🔐 Creating certificate for: ${YELLOW}$@${NC}"
+mkdir -p ${CERT_DIRECTORY}
+mkcert -key-file=${KEY_FILE} -cert-file=${CERT_FILE} ${DOMAIN}
 
-function install_ssh_certificate() {
-	KEY_NAME="${1}.key"
-	CRT_NAME="${1}.crt"
-	echo "Downloading SSL certificate $CRT_NAME from $S3_BUCKET"
-	aws ${PROFILE} s3 cp "${S3_BUCKET}${KEY_NAME}" ${DIR} 1>/dev/null
-	echo "Downloading SSL key $KEY_NAME from $S3_BUCKET"
-	aws ${PROFILE} s3 cp "${S3_BUCKET}${CRT_NAME}" ${DIR} 1>/dev/null
+echo -e "🔗 Symlinking the certificate for nginx at ${NGINX_HOME}"
+ln -sf ${KEY_FILE} ${NGINX_HOME}/${DOMAIN}.key
+ln -sf ${CERT_FILE} ${NGINX_HOME}/${DOMAIN}.crt
 
-	echo "Linking SSL certs/keys to $nginxHome"
-	sudo ln -fs "${DIR}/${CRT_NAME}" "${nginxHome}/${CRT_NAME}"
-	sudo ln -fs "${DIR}/${KEY_NAME}" "${nginxHome}/${KEY_NAME}"
-}
+echo -e "🔗 Symlinking nginx config file"
+ln -sf ${SOURCE_DIR}/frontend.conf ${NGINX_HOME}/servers/frontend.conf
 
-function install_ssh_certificate_in_jdk_ca() {
-    echo "Importing SSL certificate into Java keystore ${jdkHome}/jre/lib/security/cacerts"
-	sudo "${jdkHome}/bin/keytool" -import -alias ${1} -keystore "${jdkHome}/jre/lib/security/cacerts" -file "${1}.crt" -storepass "changeit" -noprompt
-}
-
-function install_nginx_configuration() {
-    echo "Linking Nginx configuration ${1} to $nginxHome/sites-enabled"
-    sudo ln -fs "$DIR/${1}" "$nginxHome/sites-enabled/${1}"
-}
-
-install_ssh_certificate ${SSL_CERT_NAME}
-install_ssh_certificate_in_jdk_ca ${SSL_CERT_NAME}
-install_nginx_configuration ${NGINX_SITE_CONF}
-
-echo "Restarting Nginx"
+echo -e "🚀 ${YELLOW}Restarting nginx, which requires sudo - enter password when prompted.${NC}"
 sudo nginx -s stop
 sudo nginx
 
-echo "Done. (To setup Dotcom Identity Frontend please follow identity-platform README.)"
+echo -e "💯 Done. Remember to add an entry to /etc/hosts for ${DOMAIN}."
```

Comment:
We could automate this as well? Check if the hosts file already contains the entry and append it if it does not? Something like this off the top of my head:

```bash
if grep '127.0.0.1' /etc/hosts | grep ${DOMAIN} ; then
  echo -e "✅ /etc/hosts entry already exists for ${DOMAIN}"
else
  echo -e "🔧 adding /etc/hosts entry for ${DOMAIN}"
  echo "127.0.0.1		${DOMAIN}" >> /etc/hosts
fi
```

If we're expecting this to be a consistent and required step then best to automate it.

Reply:
Updated. I've wrapped the writing to the hosts file in `sudo sh -c` as the setup script doesn't need to be (and shouldn't either as I think it causes file permission issues) executed as `sudo`.

Reply:
Ideally we would have a CNAME in DNS for this, similar to the `local.dev-gutools.co.uk` domains used in Tools. That is, solve this at the DNS level rather than faking a DNS entry per developer machine.

Reply:
> Ideally we would have a CNAME in DNS for this, similar to the local.dev-gutools.co.uk domains used in Tools. That is, solve this at the DNS level rather than faking a DNS entry per developer machine.

Agreed. `m.thegulocal.com` is leftover from a ways back and the inconsistency is a bit confusing. This is great as an interim improvement though. I've not run it to test, but conceptually looks great to me.